src/CameraControls.ts

```diff
@@ -714,22 +737,9 @@ export class CameraControls extends EventDispatcher {
 
 				if (
 					( ! this._isDragging && this._lockedPointer ) ||
-					this._isDragging && ( event.buttons & MOUSE_BUTTON.LEFT ) === MOUSE_BUTTON.LEFT
-				) {
-
-					this._state = this._state | this.mouseButtons.left;
-
-				}
-
-				if ( this._isDragging && ( event.buttons & MOUSE_BUTTON.MIDDLE ) === MOUSE_BUTTON.MIDDLE ) {
+					this._isDragging && ( event.buttons & MOUSE_BUTTON.ALL ) > 0 ) {
 
-					this._state = this._state | this.mouseButtons.middle;
-
-				}
-
-				if ( this._isDragging && ( event.buttons & MOUSE_BUTTON.RIGHT ) === MOUSE_BUTTON.RIGHT ) {
-
-					this._state = this._state | this.mouseButtons.right;
+					this._state = this._state | this.mouseEventToAction( event );
```

Comment:
I was able to simplify this conditional because of this change.

`( event.buttons & MOUSE_BUTTON.ALL ) > 0` looks a little weird, and technically it would exclude `event.buttons === 8 || event.buttons === 16`, which is not a common use case. I wanted the result of the function to match previous behaviour exactly.